doc/performance.md

```diff
@@ -0,0 +1,97 @@
+# Performance with Native Code
+
+How to assess performance of Dart and native code, and how to improve it.
+
+## Profiling Performance
+
+
+| Tool                                  | Platform  | Primary Use Case                        | Measures (Dart CPU)          | Measures (Native CPU)    | Measures (Dart Heap) | Measures (Native Heap)                                           |
+| ------------------------------------- | --------- | --------------------------------------- | ---------------------------- | ------------------------ | -------------------- | ---------------------------------------------------------------- |
+| Dart DevTools                         | All       | Profiles Dart VM, UI jank, Dart heap    | Yes                          | Opaque "Native" block    | Yes                  | Tracks "External" VM-aware memory only; Misses native-heap leaks |
+| Xcode Instruments (Time Profiler)     | iOS/macOS | Profiles native CPU call stacks         | No                           | Yes (full symbolication) | No                   | No                                                               |
+| Xcode Instruments (Leaks/Allocations) | iOS/macOS | Profiles native heap (malloc, mmap)     | No                           | No                       | No                   | Yes                                                              |
+| Android Studio Profiler (CPU)         | Android   | Profiles native C/C++ CPU execution     | No                           | Yes (traces C++ calls)   | No                   | No                                                               |
+| Perfetto (heapprofd)                  | Android   | Advanced native heap profiling          | No                           | No                       | No                   | Yes (traces malloc/free call stacks)                             |
+| Linux perf                            | Linux     | Unified Dart AOT + Native CPU profiling | Yes (requires special flags) | Yes                      | No                   | No                                                               |
+
+<!-- TODO: Add documentation for the other tools. -->
+
+### Dart DevTools
+
+For only assessing the performance of the Dart code, and treating native code as
+a black box, use the Dart performance tooling.
+
+See the documentation on https://dart.dev/tools/dart-devtools and
+https://docs.flutter.dev/perf. For FFI, most specifically, you can use
+https://docs.flutter.dev/tools/devtools/cpu-profiler and
+https://docs.flutter.dev/tools/devtools/performance#timeline-events-tab.
+For synchronous FFI calls you can add synchronous timeline events, and for
+asynchronous code (using async callbacks or helper isolates) you can use async
+events.
+
+### `perf` On Linux
+
+To see both Dart and native symbols in a flame graph, you can use `perf` on
+Linux.
+
+For JIT:
+
+```
+$ perf record -g out/DebugX64/dart-sdk/bin/dart --generate-perf-events-symbols benchmarks/FfiCall/dart/FfiCall.dart
```

Comment:
Do you have to build your own SDK to do this? Or can you use the pre-build on you have likely on your system? If you need to built your own we should call this out and link to instructions. If not, maybe we should just call this `path/to/dart` or something like that?